src/utils/utils.py

```diff
@@ -40,3 +43,29 @@ def string_to_bool(value: str) -> bool:
     if value.lower() in ["false", "0", "f", "n", "no"]:
         return False
     raise ValueError(f"{value} is not a valid boolean value.")
+
+
+def parse_k8s_token(token: str) -> Any:
+    """Decode the JWT token without verification."""
+    try:
+        return jwt.decode(token, options={"verify_signature": False})
+    except Exception as e:
+        raise ValueError("Failed to parse k8s token") from e
+
+
+def get_user_identifier_from_token(token: str) -> str:
```

Comment:
it'd be nice if you explain what it does as a part of function doc. I am bit newer to the authentication types: sub, email, service-account

Reply:
Its a [JWT token](https://jwt.io/), and the payload of a JWT token commonly contains `sub` as a unique identifier. Based on different [k8s authentication options](https://kubernetes.io/docs/reference/access-authn-authz/authentication/), I have used the keys to get the data. Also, I have verified a token from SKR and the token contains `sub` and `email`.